src/jdk.jlink/share/classes/jdk/tools/jlink/internal/ImagePluginStack.java

```diff
@@ -192,6 +192,7 @@ public ImagePluginStack(ImageBuilder imageBuilder,
 
     public void operate(ImageProvider provider) throws Exception {
         ExecutableImage img = provider.retrieve(this);
+        img.setTargetPlatform(imageBuilder.getTargetPlatform());
```

Comment:
See above comment - the ExecutableImage should know the platform when it's constructed.

Reply:
Removed the code.